.github/workflows/cigh.yml

```diff
@@ -0,0 +1,124 @@
+# This workflow aims at building and testing FreeCAD
+# It uses a runner based on latest Ubuntu LTS version
+# It builds using CMake/CCache+Make/Clang as a toolchain
+#
+# It has been foreseen to be used with CMake/Ninja/Clang...
+# ...but this is disabled at the moment because Ninja...
+# ...only works based on file timestamps, which isn't...
+# ...compatible with Git not managing them
+
+name: FCBuildNTest
+
+on:
+  workflow_dispatch: # Triggered manually
+# To run manually, use :
+# curl -X POST -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/:OWNER/FreeCAD/actions/workflows/cigh.yml/dispatches -d '{"ref":":BRANCH_NAME"}' -u :USER_NAME
+# It will then require your Github PAT to trigger the launching event
+    branches-ignore: # Do not run on master and release branches
+      - master
+      - releases/**
+  pull_request:
+    branches-ignore: # Do not run on master and release branches
+      - master
```

Comment:
No I think it will run because the PR will create a virtual branch on mainstream.
IMO this is more or less useless here, but this was guardrail that I set when doing the tests and that I'd like to keep ATM.

Reply:
> No I think it will run because the PR will create a virtual branch on mainstream.

If a contributor uses his master branch to open a PR it also comes up here as master branch.

Reply:
@wwmayer This is even worse as from GH doc, `branches-ignore` on pull requests compares to target branch. So roughly always master. :)
I removed the filters now. We can fine-tune them later, eg. ignoring all PR that won't target the master. ;)